lib/msal-common/src/client/AuthorizationCodeClient.ts

```diff
@@ -390,15 +390,25 @@ export class AuthorizationCodeClient extends BaseClient {
                 this.performanceClient
             );
 
-            const reqCnfData = await invokeAsync(
-                popTokenGenerator.generateCnf.bind(popTokenGenerator),
-                PerformanceEvents.PopTokenGenerateCnf,
-                this.logger,
-                this.performanceClient,
-                request.correlationId
-            )(request, this.logger);
+            let reqCnfData;
+            const popKid = request.popKid;
+
+            if (!popKid) {
+                const generatedReqCnfData = await invokeAsync(
+                    popTokenGenerator.generateCnf.bind(popTokenGenerator),
+                    PerformanceEvents.PopTokenGenerateCnf,
+                    this.logger,
+                    this.performanceClient,
+                    request.correlationId
+                )(request, this.logger);
+                reqCnfData = generatedReqCnfData.reqCnfString;
+                request.signPopToken = true;
+            } else {
+                reqCnfData = this.cryptoUtils.base64UrlEncode(popKid);
```

Comment:
Does this result in an encoded string from "{ kid: popKid }"?

Reply:
Thanks for catching this! Fixed.